client/src/components/DateSelector/DateSelector.js

```diff
@@ -0,0 +1,29 @@
+import React, { useState } from "react";
+import styles from "./DateSelector.module.css";
+import { PropTypes } from "prop-types";
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+
+export const DateSelector = ({ onDateChanged, showTime }) => {
```

Comment:
The prop should also be changed to `onDateChangeHandler` or something similar

Reply:
Additionally, what's the purpose of having the `handleDateChanged function as opposed to onDateChanged`? If it's a handler for the onChange date, then you can set that in the page (with the date) and pass it as the prop rather than having another "middleman" function